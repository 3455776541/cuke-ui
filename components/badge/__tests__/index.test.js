import React from "react";
import assert from "power-assert";
import { render, shallow } from "enzyme";
import toJson from "enzyme-to-json";
import Badge from "../index";

describe("<Badge/>", () => {
	it("should render a <Badge/> components", () => {
		const wrapper = render(
			<div>
				<Badge count={10}>哈哈</Badge>
			</div>
		);
		expect(toJson(wrapper)).toMatchSnapshot();
	});

	it("should find cuke-badge classnames", () => {
		const wrapper = render(
			<div>
				<Badge count={10}>哈哈</Badge>
				<Badge count={10} dot>
					哈哈
				</Badge>
			</div>
		);
		assert(wrapper.find(".cuke-badge").length >= 1);
		assert(wrapper.find(".cuke-badge-inner").length >= 1);
		assert(wrapper.find(".cuke-badge-dot").length === 1);
	});

	it("should not render inner width showZero false", () => {
		const wrapper = render(
			<div>
				<Badge count={0} showZero={false}>
					哈哈
				</Badge>
			</div>
		);
		assert(wrapper.find(".cuke-badge-inner").length === 0);
		expect(toJson(wrapper)).toMatchSnapshot();
	});
	it("should can trigger onClick event", () => {
		const onClick = jest.fn();
		const wrapper = shallow(
			<Badge count={10} onClick={onClick}>
				哈哈
			</Badge>
		);
		wrapper.find(".cuke-badge-inner").simulate("click");
		expect(onClick).toHaveBeenCalled();
	});
});
