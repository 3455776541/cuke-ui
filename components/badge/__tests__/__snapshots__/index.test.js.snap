// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Badge/> should not render inner width showZero false 1`] = `
<div>
  <div
    class="cuke-badge"
  >
    哈哈
  </div>
</div>
`;

exports[`<Badge/> should render a <Badge/> components 1`] = `
<div>
  <div
    class="cuke-badge"
  >
    <span
      class="cuke-badge-inner"
    >
      10
    </span>
    哈哈
  </div>
</div>
`;
